### PR TITLE
feat: Download CMake and build from source

### PR DESCRIPTION
**Pull request by kryptonite303**
### Description

This role will download the source code for CMake from GitHub with the version specified and `make install` it on the system.

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -1,5 +1,34 @@
...
+  stat:
+    path: /usr/local/bin/cmake
+  register: cmake
+
+- name: Run make install
+  become: true
+  make:
+    chdir: /tmp/cmake
+    target: install
+  when: not cmake.stat.exists
```
Comment (zakhark) [inline tasks/main.yml:34]: After we install it once, will we be able to install another version if needed if we use an incremental strategy (like static hosts)?

Comment (kryptonite303) [reply tasks/main.yml:34]: Good point, I'll add in the functionality to check the version

File: molecule/default/prepare.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+- hosts: rhel8
+  tasks:
+    - name: Add Artifactory repos
+      yum_repository:
+        baseurl: "{{ item.baseurl }}"
+        description: "{{ item.description }}"
+        gpgcheck: "{{ item.gpgcheck }}"
+        name: "{{ item.name }}"
+      loop:
+        - baseurl: "http://{{ lookup('env', 'ARTIFACTORY_USERNAME') }}:{{ lookup('env', 'ARTIFACTORY_PASSWORD') }}@artifactory.corp.mongodb.com/artifactory/rpm/8/AppStream/x86_64/os"  # noqa 204
```
Comment (briananthonymccarthy) [inline molecule/default/prepare.yml:11]: https?

File: molecule/default/prepare.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          description: Artifactory RHEL 8 AppStream Packages
+          gpgcheck: false
+          name: rhel8appstream
+        - baseurl: "http://{{ lookup('env', 'ARTIFACTORY_USERNAME') }}:{{ lookup('env', 'ARTIFACTORY_PASSWORD') }}@artifactory.corp.mongodb.com/artifactory/rpm/8/BaseOS/x86_64/os"  # noqa 204
```
Comment (briananthonymccarthy) [inline molecule/default/prepare.yml:15]: https?

**Pull request merged**